### ISSUE TITLE
Write Documentation for your code

### ISSUE THREAD

**Issue by daknob** [KostasMp/PortScanner#25]
When a developer wants to volunteer their time to your project, they will have no idea where your code is, how it is executed, which classes do you use, and in general they will get lost. A good starting point in these cases is to read the project's documentation. However, from what I can see, you do not seem to have any. 

Code Documentation can exist in multiple forms. The more you have the better it is for your project as the potential contributors can pick their preferred way. 

The most basic method is well-commented code. You should add a lot of comments in your code explaining everything you do, how you do it, and why. Every method and function should clearly have its arguments, its output value, as well as what it does. Finally, you should also include any contracts you may have, such as what kind of input you accept, etc. 

Another method is by having a `/doc` folder containing design documents, which will include a high level design, potentially with diagrams, etc. This folder can also contain information about every class, object, method, function, and maybe execution path of your code. This can be extremely helpful for someone trying to understand your code as well as you when debugging it. 

In any case, you need to keep the documentation and the code in sync, for example by committing both a code change and a documentation change. Feel free to check how other projects document their code and design and don't hesitate to follow their practices.

**Comment by KostasMp** [comment]
This is not a project for newbies, if you don't understand the code you should get the fvck out immediately.
This project's target group is worldwide acknowledged hackers (aka **Richard Stallman**, **John Tsiomp** etc), not script kiddies and noobs who attempt to learn coding for the first time.

**Comment by DaKnOb** [comment]
I do not think this is a nice way to welcome potential contributors. Although you have the GNU philosophy inside you.. :P

**Comment by KostasMp** [comment]
From the info i have gathered about you i can assume that you are a mac user. YOU CLOSED SOURCE IDIOTS HAVE NO PLACE IN THIS WORLD. **EVERYTHING SHOULD BE FREE INCLUDING THE RIGHTS TO KILL YOU.**
